src/set_relation/set_relation.cc

```diff
@@ -2280,6 +2518,31 @@ Set* Set::boundTupleExp(const TupleExpTerm& tuple_exp) const {
 }
 
 
+
+/* Mahdi Soltan Mohammadi
+     Projects out tuple varrable No. tvar
+     tvar is calculated based on total ariety (in+out) starting from 0.
+     Consequently, to project out jp from S: tvar = 5
+     S= {[i,j,k,ip,jp,kp] : ...}
+*/
+bool Set::project_out(int tvar)
```

Comment:
As with Relation::project_out, this logic should all be moved into a SparseConstraints::project_out so it only appears once.  Why don't you have to modify the arity for the Set here like you do in Relation?